lib/apiservers/portlayer/restapi/handlers/interaction_handlers.go

```diff
@@ -218,17 +222,66 @@ type FlushingReader struct {
 	io.Reader
 	io.WriterTo
 
-	flusher GenericFlusher
+	flusher   GenericFlusher
+	initBytes []byte
 }
 
 func NewFlushingReader(rdr io.Reader) *FlushingReader {
-	return &FlushingReader{Reader: rdr, flusher: nil}
+	return &FlushingReader{Reader: rdr, flusher: nil, initBytes: nil}
+}
+
+func NewFlushingReaderWithInitBytes(rdr io.Reader, initBytes []byte) *FlushingReader {
+	return &FlushingReader{Reader: rdr, flusher: nil, initBytes: initBytes}
 }
 
 func (d *FlushingReader) AddFlusher(flusher GenericFlusher) {
 	d.flusher = flusher
 }
 
+// readDetectInit() is used by WriteTo() which is used by io.Copy.  It attempts
+// to detect a init byte buffer.  If it finds that init byte sequence, it is
+// ignored.  This reader does not care about the init sequeunce.  The init sequence
+// maybe used by the higher level interaction, which in this case is the Swagger
+// establishing initial connection for stdin.
+func (d *FlushingReader) readDetectInit(buf []byte) (int, error) {
+	nr, er := d.Read(buf)
+	if d.initBytes != nil {
+		initLen := len(d.initBytes)
+		total := nr
+		for {
+			if total < initLen {
+				if er != nil {
+					// Got an error, zero out len and break out to return error back
+					nr = 0
+					break
+				}
+				nr, er = d.Read(buf[total:])
+				total += nr
```

Comment:
This is going to prevent us from ever sending `< len(d.initBytes)` as the first input.

This is written with an assumption that the other end of the data channel is sending the correct data, with basic precautions. We can either:
a. assume that the source is aware of how to behave and simply discard the first X bytes
b. code defensively in the expectation that it doesn't.

In the latter case we should be comparing what we read in each iteration of the loop as that allows us to detect, in a timely fashion, unexpected usage.